prog/prog.go

```diff
@@ -116,6 +116,14 @@ func (a *Arg) Size() uintptr {
 				size += fld.Size()
 			}
 		}
+		align := typ.Align()
+		if size % align != 0 {
```

Comment:
It would need to be:
``` go
align := typ.Align()
if size%align != 0 && typ.Varlen {
	size += align - size%align
}
if size%align != 0 {
	panic(fmt.Sprintf("struct %+v with type %+v has static size %v, which isn't aligned to %v", a, typ, size, align))
}
```
I'm not sure if this looks better than what I have now.